pkg/cmd/registry/delete/delete.go

```diff
@@ -142,17 +144,19 @@ func runDelete(opts *options) error {
 
 	opts.Logger.Info(icon.SuccessPrefix(), opts.localizer.MustLocalize("registry.delete.log.info.deleteSuccess", localize.NewEntry("Name", registryName)))
 
-	currentContextRegistry := cfg.Services.ServiceRegistry
+	currentContextRegistry := currCtx.ServiceRegistryID
 	// this is not the current cluster, our work here is done
-	if currentContextRegistry == nil || currentContextRegistry.InstanceID != opts.id {
+	if currentContextRegistry != opts.id {
 		return nil
 	}
 
-	// the service that was deleted is set as the user's current cluster
-	// since it was deleted it should be removed from the config
-	cfg.Services.ServiceRegistry = nil
-	err = opts.Config.Save(cfg)
-	if err != nil {
+	// the service that was deleted is set as the user's current instance
+	// since it was deleted it should be removed from the context
+
+	currCtx.ServiceRegistryID = ""
```

Comment:
This seems to be too low level. Can we actually have method that updates and saves context. 
I think we expose too much details here

Reply:
I meant this line and the next 4 lines that save context